src/nextjournal/clerk/hashing.clj

```diff
@@ -237,9 +238,18 @@
               deps)
       state-with-var-hash)))
 
+(defn analyze-doc
+  ([doc]
+   (analyze-doc {:graph (dep/graph)} doc))
+  ([state doc]
+   (let [state-with-document (assoc state :doc doc)
+         code-cells (into [] (filter (comp #{:code} :type)) (:doc doc))]
```

Comment:
Probably worth lifting `(into [] (filter (comp #{:code} :type)) (:doc doc))` into a helper and calling both here and from `analyze-file`

Reply:
should `analyze-doc` ever care about the `:markdown?` option? If so it might be worth writing `analyze-file` read the file and pass it to `analyze-doc` for doing work on